regparser/tree/xml_parser/paragraph_processor.py

```diff
@@ -147,21 +160,56 @@ class TableMatcher(object):
     def matches(self, xml):
         return xml.tag == 'GPOTABLE'
 
-    def derive_nodes(self, xml):
+    def derive_nodes(self, xml, processor=None):
         return [Node(table_xml_to_plaintext(xml), label=[mtypes.MARKERLESS],
                      source_xml=xml)]
 
 
+class HeaderMatcher(object):
+    def matches(self, xml):
+        return xml.tag == "HD"
+
+    def derive_nodes(self, xml, processor=None):
+        # This should match HD elements only at lower levels, and for now we'll
+        # just put them into the titles
+        return [Node(text='', title=tree_utils.get_node_text(xml).strip(),
+                     label=[mtypes.MARKERLESS])]
+
+
+class ExtractMatcher(object):
+    def matches(self, xml):
+        return xml.tag in ('EXTRACT')
+
+    def derive_nodes(self, xml, processor=None):
+        extract_node = Node(text=xml.text, node_type=u"extract",
```

Comment:
Mind `(xml.text or '').strip()` and adding an EXTRACT node type to `regparser.tree.struct`?